collectors/python.d.plugin/dns_query_time/dns_query_time.chart.py

```diff
@@ -89,13 +84,15 @@ def dns_req(ns, t, q):
         request = dns.message.make_query(domain, dns.rdatatype.A)
 
         try:
-            dns_start = time()
-            dns.query.udp(request, ns, timeout=t)
-            dns_end = time()
-            query_time = round((dns_end - dns_start) * 1000)
-            q.put({'_'.join(['ns', ns.replace('.', '_')]): query_time})
+            resp = dns.query.udp(request, ns, timeout=t)
+            if (resp.rcode() == dns.rcode.NOERROR and resp.answer):
```

Comment:
wait, `rcode` is `int` according docs

http://www.dnspython.org/docs/1.14.0/dns.message.Message-class.html

how is it `rcode()` ?

Reply:
wait,  it is method, i am wrong

Reply:
I dunno, but if you try it you'll see:

```import dns.query

qname = dns.name.from_text('www.google.com')
q = dns.message.make_query(qname, dns.rdatatype.A)
resp = dns.query.udp(q, '8.8.8.8')
print(resp.rcode == dns.rcode.NOERROR)
print(resp.rcode() == dns.rcode.NOERROR)
```

Reply:
Sorry for the mess. My bad.

Reply:
not a problem - thanks for working through and offering advice!